### ISSUE TITLE
ImportError: cannot import name 'mail'

### ISSUE THREAD

**Issue by Davidriquelme** [miguelgrinberg/flasky#345]
Hi, I know this is very common error, but can´t seem to find a solution for my dev environment.

so I created the object and then initialized
![image](https://user-images.githubusercontent.com/12628676/37994605-6c9b876c-31e8-11e8-967f-9fd47673d9f0.png)

and the import wich is causing the error:
![image](https://user-images.githubusercontent.com/12628676/37994909-4cf753cc-31e9-11e8-9435-21866f9fdb6d.png)

and the error:
![image](https://user-images.githubusercontent.com/12628676/37994720-c3e8e870-31e8-11e8-8a7b-5a4f017aa96b.png)

any thoughts ? Pycharm has not thrown me any errors

pd: I haven't touch the manage.py file.
![image](https://user-images.githubusercontent.com/12628676/37994768-e6e0abba-31e8-11e8-8a9c-b3481fa6d321.png)

**Comment by miguelgrinberg** [comment]
Yes, this is very common. It's a circular dependency problem. If you look in your stack trace, the last appearance of `app/__init__.py` shows line 6, the import from `app.api`. This line is above the place where `mail` is defined, so at this point `mail` does not exist yet.

Following the stack trace, `app.api` imports `app.api.views`, which imports `app.api.endpoints`, which imports `app.api.endpoints.send_mail`, which imports `app.mail`, and obviously errors because the app package is still running at line 6.

Possible solution: move the offending import inside the `send_mail()` function, so that the import gets delayed until after the app package is fully parsed.

**Comment by Davidriquelme** [comment]
thanks you so much Miguel, I'm gonna try what you said and come back with a solution.

**Comment by Davidriquelme** [comment]
so I move the "import" inside the send_mail(), and it works just fine. 
![image](https://user-images.githubusercontent.com/12628676/38028693-105ca828-326a-11e8-9476-82e7a00d3a59.png)


thanks @miguelgrinberg
